src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -79,9 +79,7 @@ CoreWorkerPlasmaStoreProvider::~CoreWorkerPlasmaStoreProvider() {
 
 Status CoreWorkerPlasmaStoreProvider::SetClientOptions(std::string name,
                                                        int64_t limit_bytes) {
-  std::lock_guard<std::mutex> guard(store_client_mutex_);
-  RAY_RETURN_NOT_OK(store_client_.SetClientOptions(name, limit_bytes));
-  return Status::OK();
+  return store_client_.SetClientOptions(name, limit_bytes);
```

Comment:
Don't we still need a lock?

Reply:
I think we already have `std::lock_guard<std::recursive_mutex> guard(client_mutex_);` in all store client function calls.

Reply:
Sounds good. Thanks for the confirmation